### PR TITLE
Reword documentation section of readme and fix links.

### LINKED ISSUE #1692: Broken docs link
Hi @Matthew-Jennings,

The following link is broken on the GitHub README page:

image

Might you be in a position to fix that?

### PR DESCRIPTION
**Pull request by Matthew-Jennings**
Closes #1692

### CODE REVIEW

**Comment by Matthew-Jennings** [comment]
Unless someone else is cancelling the tests, it appears that the `pyright` installation failure may be causing the other tests to auto-cancel. This failure appears to be related to the `monaco` tests

I can't see an obvious way around this other than A: waiting for the monaco tests to be fixed or B: manually re-running all of the non-pyright tests. The only way I know how to do the latter is *very* tedious because each item can only be re-run one-at-time 😞  I'd really rather avoid that.

File: README.rst
Comment (sjswerdloff) [inline README.rst:79]: This might be confusing, because there aren't any links or descriptive text that says Contributors Guide, instead there is a link entitled "workstation setup guides".

Comment (Matthew-Jennings) [reply README.rst:79]: Yeah. I guess this is a result of the transition to the new sturcture still being a work-in-progress (see the bottom of the opening post of #1665 and the pages that are listed in the proposed new structure that do not yet exist). 

For now, the side bar (see below) at least matches the categorisation implied by my new wording. Perhaps the bolded text could be converted into a link to the [Overview ](https://docs.pymedphys.com/en/latest/contrib/index.html) page under the Contributors Guide sidebar heading. Unfortunately, this page isn't yet the "intro to contributing" that it ought to be. [I think this was something that @nlambriICH was interested in working on](https://github.com/pymedphys/pymedphys/issues/1665#issuecomment-1153734521), but that may no longer be the case.

image

Comment (Matthew-Jennings) [reply README.rst:79]: FWIW, the `docs` folder structure also follows this overarching categorisation. 
image

Comment (nlambriICH) [reply README.rst:79]: I'm still interested to rework that part of the documentation, but I didn't find time to work on it... @ddcc3388 has offered to help out (#1588)

**Comment by SimonBiggs** [comment]
> Unless someone else is cancelling the tests, it appears that the `pyright` installation failure may be causing the other tests to auto-cancel. This failure appears to be related to the `monaco` tests

A few months ago I set up the tests to auto-cancel if one fails. It's not ideal, but we were often hitting scenarios where we had no CI machines available and test were taking ages.

I have fixed those pyright tests over at https://github.com/pymedphys/pymedphys/pull/1702. So once that's merged, that should help here.

**Comment by guanfada** [comment]
@Matthew-Jennings @SimonBiggs @nlambriICH 
I believe we have addressed most of the reviewer's concerns. Next, who can help on the section of "automated tests"?

File: README.rst
Comment (Matthew-Jennings) [reply README.rst:155]: Great!

**Comment by Matthew-Jennings** [comment]
> A few months ago I set up the tests to auto-cancel if one fails. It's not ideal, but we were often hitting scenarios where we had no CI machines available and tests were taking ages.

Make sense!

**Comment by Matthew-Jennings** [comment]
> I believe we have addressed most of the reviewer's concerns. Next, who can help on the section of "automated tests"?

I'd just like to get @sjswerdloff's approval before merging. @sjswerdloff, if you're still uncomfortable with the wording, would you suggest an alternative?

**Comment by SimonBiggs** [comment]
@Matthew-Jennings, if the current links on the readme don't work, isn't that a problem with our re-directs? Didn't we aim to make it so that all old links appropriately auto-forward through to the new documentation?

**Comment by Matthew-Jennings** [comment]
Yes, good point!

Unfortunately, it looks like none of the redirects are working because readthedocs is applying the "Prefix Redirect" before applying the "Page Redirect":

image

I can try to find a good solution for this in the next couple of days. In the meantime, it seems to me that it is probably still worth merging this in. The README is now starting to more closely describes the (intended) docs structure and it's probably better practice to use updated links throughout the PyMedPhys docs. What do you think?

**Comment by SimonBiggs** [comment]
> What do you think?

I certainly agree :)

**Comment by SimonBiggs** [comment]
I'll review and merge after lunch

**Comment by Matthew-Jennings** [comment]
> I'll review and merge after lunch.
> 
> [Edit]
> 
> Actually, I just saw that we've opted to wait for @sjswerdloff for this one. If you want my review, let me know and I'll review and merge. No need to have us both be assigned for review though. No need to double up work.

Yes, sorry. I just meant to get your opinion on whether we should fix the links in this way at all (i.e., fix the readthedocs issue _only_).

**Comment by SimonBiggs** [comment]
> Yes, sorry. I just meant to get your opinions on:
> 
> 1. whether we should fix the links in this way at all (i.e., fix the readthedocs issue _only_).
> 2. whether this should be merged if the readthedocs issue hasn't been fixed.

Cheers :slightly_smiling_face: 

As a general rule which I think would be safe in applying without needing to check in. If a fix/feature can easily be done in another PR, and without that fix/feature this PR still leaves the repository in a better state than without it, then safe to merge.

I look at this as a corollary of our golden rule for code review approval:

> In general, reviewers should favor approving a PR once it is in a state where it definitely improves the overall code health of the system being worked on, even if the PR isn’t perfect.

-- https://google.github.io/eng-practices/review/reviewer/standard.html

File: README.rst
Comment (SimonBiggs) [inline README.rst:40]: Do we really need this in our readme file?

File: README.rst
Comment (SimonBiggs) [inline README.rst:278]: Might we be able to get these link fixes in a separate PR and then just merge that? That way the bug fix won't be tied to the other changes in this PR

File: README.rst
Comment (crcrewso) [reply README.rst:40]: I don't think we need the PyPi link as the current version is autogenerated in the header
I'm unsure about github
Either way it's not hurting anything and the descriptions make sense so either way works for me

Comment (sjswerdloff) [reply README.rst:40]: I'm a strong believer in pointing to the source code (Github for PyMedPhys).
I benefitted from gcc and gdb back in the day, and had to hack it to work on HP-UX 6.5...
So... If someone uses pip to install PyMedPhys, will the readme be available and point them in the right direction to find the source?
If not, where can the user determine where to find the source?  I realise that the source is in fact embedded in the package, but finding fixes that are not yet released is one reason for hunting down the repo...

**Comment by sjswerdloff** [comment]
> > I believe we have addressed most of the reviewer's concerns. Next, who can help on the section of "automated tests"?
> 
> I'd just like to get @sjswerdloff's approval before merging. @sjswerdloff, if you're still uncomfortable with the wording, would you suggest an alternative?

I'm happy: I approve.  Do you want or need me to formally review and approve now?

File: README.rst
Comment (Matthew-Jennings) [reply README.rst:278]: I'd really rather not. Some of the other changes here have already been reviewed and, honestly, there really isn't much here!

It has been nearly two weeks since I replied to the last review. @sjswerdloff, are you please able to re-review asap? Otherwise, @SimonBiggs or @pchlap: perhaps one of you could instead make a call given the timeframe?

File: README.rst
Comment (SimonBiggs) [reply README.rst:40]: Sorry, I was pointing to the long "statement of need" (see the first line of green).

@Matthew-Jennings, do we really need this whole statement of need here?

**Review by sjswerdloff — APPROVED** [review]
I'm OK with this for now.
I think we should merge, and then revisit any remaining comments in a new issue +PR

**Comment by Matthew-Jennings** [comment]
Thanks @sjswerdloff!

**Comment by SimonBiggs** [comment]
Thanks for sorting these out @Matthew-Jennings

**Pull request merged**
